### DISCUSSION TITLE
Chromatic aberration and discoloration after upgrading to version 1.17

### DISCUSSION THREAD

**Discussion by heakey** [Acly/krita-ai-diffusion#697] [General]
首先感谢您对这个应用程序的奉献！我经常使用局部重绘进行图像修复，但今日更新后出现了重绘区域变色问题和色差问题。

我的英文不好，请原谅我使用中文提问然后使用翻译软件进行翻译。
今日在使用Krita前进行了升级，而后发现实时绘画会出现色差问题。
无论如何改变模型、VAE和步数，仍然会出现色差。
最后我将ComfyUI和krita-ai-diffusion都回退后问题得到解决。
我不知道问题具体出在哪里，但我可以解释进行了哪些操作，以供大家分析讨论。

①将ComfyUI的版本升级到5月9日最新，版本id为：cd07340
②更新了ComfyUI的可更新扩展插件，由于使用整合包一键更新，所以无法确定具体更新了哪些扩展，抱歉。
③进行如上更新后我如往常一样打开krita软件，但发现无法使用krita-ai-diffusion，提示我ipadapter需要更新。
④此时我的krita-ai-diffusion版本为1.15，通过在网上搜索后发现需要更新版本才能不报错。所以我将版本升级为1.17.
④进行以上两次升级后，我可以正常使用了，但使用后发现出现了变色和色差问题。之前无论所选的重绘区域有多小、所选区域是否为纯色区域，都没有出现这个问题，但升级后出现了。
⑤经过反复研究无果，我最终选择将ComfyUI的版本与插件更新退回（好在我之前存有旧版本的安装包），然后将krita-ai-diffusion版本退回到1.15，问题解决、

※而且发现在升级到1.17版本后，实际重绘的步数与我设置的不一致，例如我选择LCM，步数8，但实际重绘时后台显示步数为4。

不知道大家是否遇到了同样的问题，如何解决？提前感谢。

First of all thank you for your dedication to this application! I often use local redraw for image repair, but after today's update there were discoloration issues and chromatic aberration issues in the redrawn area.

My English is not good, please forgive me for asking questions in Chinese and then using translation software for translation.
Today, before using Krita, we upgraded and found that there was a color difference problem in real-time painting.
No matter how you change the model, VAE, and number of steps, chromatic aberration will still occur.
Finally, I solved the problem by backstepping both ComfyUI and krita-ai-diffusion.
I don't know exactly what the problem is, but I can explain what has been done for everyone to analyze and discuss.

① Upgrade the ComfyUI version to the latest version on May 9, the version id is cd07340
② Updated the ComfyUI updatable extension plug-in, due to the use of integration package one-click update, so can not determine the specific update of the extension, sorry.
③ After the above update, I opened the krita software as usual, but found that I could not use Krita-ai-diffusion, indicating that the ipadapter needed to be updated.
④ At this time, my krita-ai-diffusion version is 1.15, and I find that I need to update the version after searching on the Internet. So I upgraded the version to 1.17.
④ After the above two upgrades, I can use it normally, but I found that there are discoloration and color difference problems after use. This issue didn't occur before, regardless of how small the redrawn area was or whether the area was a solid color area, but it did after the upgrade.
After repeated research, I finally chose to return the ComfyUI version and plug-in update (fortunately, I have the old version of the installation package), and then returned the krita-ai-diffusion version to 1.15, the problem is solved.

※ And found that after upgrading to version 1.17, the actual number of redrawn steps is not consistent with my setting, for example, I select LCM, the number of steps is 8, but the actual redrawn background shows the number of steps is 4.

I wonder if you have encountered the same problem, how to solve it? Thanks in advance.

**Comment by Acly** [comment]
There's a lot of discussion about similar issues in #662 

If you can provide a .kra file with the active selection that shows the problem, it would be much easier to investigate.

It could just be the 4 steps, or related to differential diffusion, (or something else), but I'm not sure yet.
